docs/thirdparty-rest-v1.0-OpenApi.yaml

```diff
@@ -0,0 +1,846 @@
+            { 
+              "id": "456",
+              "initiatorId": "PISP A",
+              "authChannels": [ "Web", "OTP" ],
+              "scopes": [ "accounts.getBalance", "accounts.list", "accounts.transfer" ],
+              "callbackUri": "pisp-app://callback..."
+            }
+      responses:
+        202:
+          $ref: '#/responses/Response202'
+        400:
+          $ref: '#/responses/ErrorResponse400'
+        401:
+          $ref: '#/responses/ErrorResponse401'
+        403:
+          $ref: '#/responses/ErrorResponse403'
+        404:
+          $ref: '#/responses/ErrorResponse404'
+        405:
+          $ref: '#/responses/ErrorResponse405'
+        406:
+          $ref: '#/responses/ErrorResponse406'
+        501:
+          $ref: '#/responses/ErrorResponse501'
+        503:
+          $ref: '#/responses/ErrorResponse503'
+
+  /consentRequests/{ID}:      
+    put:
+      description: |
+        Update a consent request
+        - Called by a PISP to add the authToken parameter
+        - Called by a DFSP to include the authorizationUri
+      summary: UpdateConsentRequest
+      tags:
+        - consentRequest
+      operationId: UpdateConsentRequest
+      produces:
+        - application/json
+      parameters:
+        - $ref: '#/parameters/ID'
+        - name: body
+          in: body
+          required: true
+          schema: 
+            $ref: '#/definitions/ConsentRequest'
+        # Headers
+        - $ref: '#/parameters/Content-Length'
+        - $ref: '#/parameters/Content-Type'
+        - $ref: '#/parameters/Date'
+        - $ref: '#/parameters/X-Forwarded-For'
+        - $ref: '#/parameters/FSPIOP-Source'
+        - $ref: '#/parameters/FSPIOP-Destination'
+        # TODO: do we need encryption on this?
+        - $ref: '#/parameters/FSPIOP-Encryption'
+        - $ref: '#/parameters/FSPIOP-Signature'
+        - $ref: '#/parameters/FSPIOP-URI'
+        - $ref: '#/parameters/FSPIOP-HTTP-Method'
+      x-examples:
+          "application/json":
+            { 
+              "id": "456",
+              "initiatorId": "PISP A",
+              "authChannels": [ "Web" ],
+              "scopes": [ "accounts.getBalance", "accounts.list", "accounts.transfer" ],
+              "callbackUri": "pisp-app://callback...",
+              "authorizationUri": 'https://dfsp_site.com/',
+              "authToken": "ABCD_1234"
+            }
+      responses:
+        202:
+          $ref: '#/responses/Response202'
+        400:
+          $ref: '#/responses/ErrorResponse400'
+        401:
+          $ref: '#/responses/ErrorResponse401'
+        403:
+          $ref: '#/responses/ErrorResponse403'
+        404:
+          $ref: '#/responses/ErrorResponse404'
+        405:
+          $ref: '#/responses/ErrorResponse405'
+        406:
+          $ref: '#/responses/ErrorResponse406'
+        501:
+          $ref: '#/responses/ErrorResponse501'
+        503:
+          $ref: '#/responses/ErrorResponse503'
+
+definitions:
```

Comment:
can we keep often used definitions in external files ?
and use them via path specification in $ref ? 
`$ref: './definitions.json#/CorrelationId'`

Reply:
or maybe it would be better to use `yaml` files instead `json` ;-)